SWLOR.Game.Server/Feature/AbilityDefinition/Force/ForceBurstAbilityDefinition.cs

```diff
@@ -26,53 +27,70 @@ public class ForceBurstAbilityDefinition : IAbilityListDefinition
+                {
+                    var attackerStat = GetAbilityScore(activator, AbilityType.Willpower);
+                    var defense = Stat.GetDefense(target, CombatDamageType.Force, AbilityType.Willpower);
+                    var defenderStat = GetAbilityScore(target, AbilityType.Willpower);
+                    var attack = Stat.GetAttack(activator, AbilityType.Willpower, SkillType.Force);
+                    var damage = Combat.CalculateDamage(
+                        attack,
+                        dmg,
+                        attackerStat,
+                        defense,
+                        defenderStat,
+                        0);
+                    var delay = GetDistanceBetweenLocations(GetLocation(activator), targetLocation) / 18.0f + 0.35f;
 
-            DelayCommand(delay, () =>
-            {
-                ApplyEffectToObject(DurationType.Instant, EffectDamage(damage), target);
-                ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.Vfx_Imp_Silence), target);
-                ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.VFX_IMP_KIN_L), target);
-            });
-            
-            Enmity.ModifyEnmity(activator, target, level * 150 + damage);
-            CombatPoint.AddCombatPoint(activator, target, SkillType.Force, 3);
+                    var dTarget = creature;
+                    AssignCommand(activator, () =>
+                    {
+                        PlaySound("plr_force_blast");
+                        DoFireball(target);
+                    });
+                    DelayCommand(delay, () =>
+                    {
+                    ApplyEffectToObject(DurationType.Instant, EffectDamage(damage), dTarget);
+                    ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.Vfx_Imp_Silence), dTarget);
+                    ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.VFX_IMP_KIN_L), dTarget);
+                    ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.Vfx_Imp_Pulse_Wind), target);
+                    });
+                    DelayCommand(delay + 0.1f, () =>
+                    {
+                        ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.Vfx_Imp_Pulse_Wind), target);
+                    });
+                    DelayCommand(delay + 0.2f, () =>
+                    {
+                        ApplyEffectToObject(DurationType.Instant, EffectVisualEffect(VisualEffect.Vfx_Imp_Pulse_Wind), target);
+                    });
+
+                    CombatPoint.AddCombatPoint(activator, creature, SkillType.Force, 3);
+                    Enmity.ModifyEnmity(activator, creature, 250 * level + damage);
+                }
+                creature = GetNextObjectInShape(Shape.Sphere, RadiusSize.Medium, GetLocation(target), true, ObjectType.Creature);
```

Comment:
This can cause an infinite loop if the if statement above is never met. Please move line 92 outside of this condition.